libraries/tock-cells/src/take_cell.rs

```diff
@@ -23,7 +23,7 @@ impl<'a, T: ?Sized> TakeCell<'a, T> {
     }
 
     /// Creates a new `TakeCell` containing `value`
-    pub fn new(value: &'a mut T) -> TakeCell<'a, T> {
+    pub const fn new(value: &'a mut T) -> TakeCell<'a, T> {
```

Comment:
This could probably be split out as it's own PR.

Reply:
Right. I've opened #2214 accordingly, whose merge commit I'll cherry-pick onto this branch such that this isn't part of the merge diff anymore.